### DISCUSSION TITLE
Giscus displays dark background in light mode

### DISCUSSION THREAD

**Discussion by SohamTilekar** [squidfunk/mkdocs-material#6938] [Q&A] [answered]
### Bug description

The Comment Section Dose Not Render properly when using python 3.11.
I Have tested it on windows + edge.

Python 3.12 [fine]: -
    Dark:
![image](https://github.com/squidfunk/mkdocs-material/assets/152298897/a29e992f-5bf0-4b2c-8fcf-d3a2e10ee103)
    Light:
![image](https://github.com/squidfunk/mkdocs-material/assets/152298897/d9344214-e698-4172-a316-05bb05d299b6)
Python 3.11 [Problem]: -
    Dark:
![image](https://github.com/squidfunk/mkdocs-material/assets/152298897/964fd906-426b-4ed8-864c-b60bb3485ee6)
    Light:
![image](https://github.com/squidfunk/mkdocs-material/assets/152298897/52db2f84-38d6-418e-8fd5-366f97094de6)

### Related links

- [Comments](https://squidfunk.github.io/mkdocs-material/setup/adding-a-comment-system/)

### Reproduction

Project Zip File :- [dump.zip](https://github.com/squidfunk/mkdocs-material/files/14652108/dump.zip)

### Steps to reproduce

1. Making the Project.
2. Just setting comments as provided in the docs. Without {if blocks}

### Browser

Edge

### Before submitting

- [X] I have read and followed the [bug reporting guidelines](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/).
- [X] I have attached links to [the documentation](https://squidfunk.github.io/mkdocs-material/), and possibly related [issues](https://github.com/squidfunk/mkdocs-material/issues) and [discussions](https://github.com/squidfunk/mkdocs-material/discussions).
- [X] I assure that I have [removed all customizations](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/#remove-customizations) before submitting this bug report.
- [X] I have attached a __.zip file__ with a [minimal reproduction](https://squidfunk.github.io/mkdocs-material/guides/creating-a-reproduction/) using the [built-in info plugin](https://squidfunk.github.io/mkdocs-material/plugins/info/).

**Comment by alexvoss** [comment]
Thanks for getting in touch. Can you please do us the favour and create the reproduction using the info plugin as described in [Creating a reproduction](https://squidfunk.github.io/mkdocs-material/guides/creating-a-reproduction/) as this packs up  additional information.

I am somewhat surprised that you tracked this down to the difference between Python 3.11 and 3.12. Are you sure that you can reproduce the desired as well as the undesired behavior simply be switching the version of the Python interpreter? I am asking because this is more likely to be an issue with the JavaScript and the browser.

Running your reproduction, I could not see the issue with Python 3.11 on MacOS and Firefox but that does not have to mean anything except I need to try again using a Windows machine a bit later on.

**Comment by squidfunk** [comment]
Thanks for reporting. Note that our touch point with Giscus is only the [`comments`](https://github.com/squidfunk/mkdocs-material/blob/master/src/templates/partials/comments.html) partial where you add the snippet that you generated on Giscus' site – nothing more. Everything else is provided by Giscus. [Our documentation also states that it's customization](https://squidfunk.github.io/mkdocs-material/setup/adding-a-comment-system/#customization), so I'm converting this to a discussion, asking for help.

**Comment by stalomeow** [comment, marked as answer]
This is caused by [the `color-scheme: light` property of the `.md-typeset iframe`](https://github.com/squidfunk/mkdocs-material/blob/1b2d48455d26ea51c4f0648cc67c37814f84d55a/src/templates/assets/stylesheets/main/_typeset.scss#L429). If you disable it, the weird black background disappears.

![Snipaste_2024-03-22_21-49-49](https://github.com/squidfunk/mkdocs-material/assets/47203031/b8297067-b873-4d64-9199-0e3f22ac0464)

Currently you can [add an additional css file](https://squidfunk.github.io/mkdocs-material/customization/?h=css#additional-css) containing the following code to temporarily fix this issue.

``` css
.md-typeset iframe.giscus-frame {
    color-scheme: light dark !important;
}
```

**Reply by SohamTilekar** [reply]
Thanks for Giving the Solution.

**Reply by squidfunk** [reply]
Hmm, this is indeed a side effect of the changes we made to fix #6889. We're going to look into it again.

**Reply by squidfunk** [reply]
This should be fixed with https://github.com/squidfunk/mkdocs-material/issues/6889#issuecomment-2016232502

**Reply by squidfunk** [reply]
The fix was released as part of [9.5.15](https://squidfunk.github.io/mkdocs-material/changelog/#9.5.15).
